test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -838,4 +838,33 @@ TEST(ClusterManagerInitHelper, AddSecondaryAfterSecondaryInit) {
   cluster2.initialize_callback_();
 }
 
+TEST(ClusterManagerInitHelper, RemoveClusterWithinInitLoop) {
+  InSequence s;
+  ClusterManagerInitHelper init_helper;
+
+  // Tests the scenario encountered in Issue 903: The cluster was removed from
+  // the secondary init list while traversing the list.
+  NiceMock<MockCluster> cluster;
+  // Fake that we are in the secondary InitializePhase for this test.
+  ON_CALL(cluster, initializePhase()).WillByDefault(Return(Cluster::InitializePhase::Secondary));
+  // We expect that initialize() should be called first upon addCluster.  In the
```

Comment:
OK, thanks! This is definitely useful.  I will ensure the rejiggered test fails with the original code, and passes with the fix.